### PR TITLE
Add wrap boundary code tests

### PR DESCRIPTION
**Pull request by leynos**
## Summary
- avoid splitting inline code spans when wrapping
- extend helper to check that code spans stay intact
- test with a three-digit list prefix
- verify inline code wrapping at boundary

## Testing
- `cargo clippy -- -D warnings`
- `RUSTFLAGS="-D warnings" cargo test`

------
https://chatgpt.com/codex/tasks/task_e_68744fcc1640832299e4061e2e2a2bdf

## Summary by Sourcery

Implement custom token-based wrapping to preserve inline code spans and integrate it into paragraph and list formatting, while adding tests for three-digit list prefixes and inline code boundary conditions.

New Features:
- Introduce wrap_preserving_code function for token-based text wrapping that preserves inline code spans.

Bug Fixes:
- Prevent inline code spans from being split across lines during wrapping.

Enhancements:
- Replace textwrap fill usage with wrap_preserving_code in paragraph and list wrapping logic.
- Add TOKEN_RE regex to support tokenization of code spans, words, and whitespace.

Tests:
- Extend assert_wrapped_list_item helper to verify code spans remain intact.
- Add tests for wrapping list items with three-digit prefixes and for inline code boundary wrapping.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -241,6 +290,39 @@ static THEMATIC_BREAK_RE: std::sync::LazyLock<Regex> = std::sync::LazyLock::new(
...
+
+        if !token.chars().all(char::is_whitespace) {
+            current.push_str(&token);
+            current_width = token_width;
+        }
+    }
+    let trimmed = current.trim_end();
+    if !trimmed.is_empty() {
+        lines.push(trimmed.to_string());
+    }
```
[the remark this answers is not in the document]
Comment (leynos) [reply src/lib.rs:322]: Please create a GitHub issue for this

File: src/lib.rs [hunk 2]
```diff
@@ -241,6 +290,39 @@ static THEMATIC_BREAK_RE: std::sync::LazyLock<Regex> = std::sync::LazyLock::new(
...
+            lines.push(trimmed.to_string());
+        }
+        current.clear();
+        current_width = 0;
+
+        if !token.chars().all(char::is_whitespace) {
+            current.push_str(&token);
+            current_width = token_width;
+        }
+    }
```
[the remark this answers is not in the document]
Comment (leynos) [reply src/lib.rs:318]: Please create a GitHub issue for this
